airbyte-integrations/connectors/source-s3/source_s3/source_files_abstract/formats/jsonl_parser.py

```diff
@@ -45,8 +57,12 @@ def _parse_options(self, json_schema: Mapping[str, Any] = None) -> Mapping[str,
             "unexpected_field_behavior": self.format.unexpected_field_behavior,
         }
         if json_schema:
-            parse_options["explicit_schema"] = pa.schema(self.json_schema_to_pyarrow_schema(json_schema))
-
+            schema = self.json_schema_to_pyarrow_schema(json_schema)
+            schema = pa.schema({
+                field: type_ for field, type_ in schema.items()
+                if type_ not in self.NON_SCALAR_TYPES.values()
```

Comment:
Why did you decide to remove the `struct` from the schema? Your PR description suggests the nested field will still be processed, is it right? Will they still be part of the stream schema?

Reply:
I remove `struct`s from schema only when the schema is passed to the reader so the data is read fully and missing schema parts are inferred by the reader. This is done because the reader accepts only valid pyArrow schemas. 
Let's consider a use case:
  - File has a schema like `{'a': 'integer', 'b': 'boolean', 'c': 'struct<...>'}`
  - User provides a schema {'a': integer, 'b': 'boolean', 'c': 'object', 'd': 'string'}
  - The connector needs to merge these schemas first. To do it, pyarrow schemas are converted into json schemas
  - Since user can not provide an object schema for field `c`, no need to cope with pyarrow's `struct` structure
  - When merging, we get `{...'c': 'object'...}`
  - Then master schema is passed to a reader(pyarrow) so it could actually read the file
  - Pyarrow does not accept json schemas, so master schema has to be converted back to pyarrow datatypes. Empty `struct` type can not be instantiated as well as passed in as passed in. Also, we do not have info about detailed structure of this struct/object. That's why we exclude `struct`s from schemas when passing them to pyarrow
  - pyarrow reads all the data, type of missing fields in schema is inferred

Each `struct` will remain at the discovery stage as well as when merging schemas into master schema.

Reply:
Thank you for the detailed explanation. And what happens if 'c' field is a `struct` in the first file and is an `int` in the second file? Or the other way around?

Reply:
I think the answer is here: https://github.com/airbytehq/airbyte/pull/16631 😄

Reply:
that's right :)